### ISSUE TITLE
Potential memory leak in opening RStudio Connect dialog

### ISSUE THREAD

**Issue by fereshtehRS** [rstudio/rstudio#1924]
RStudio Version 1.1.412 on MacOS 10.11.5

I was testing an integration issue between IDE and RStudio Connect Server, and afterwards noticed that the memory footprint of `RStudio` process is pretty high (closed to 2G).

Restarted IDE and was able to get it to that point following these steps:

- Pick an RSC that you can connect to
- Create or select a user that has `Viewer` role
- Access IDE menu Tools -> Global Options -> Publishing 
- Click on `Connect` and then  `RStudio Connect`
- Enter valid RSC url and Next
- RStudio Connect dialog opens
1- Enter credentials for the `Viewer` user and Login
2- You get message about ` insufficient privileges to publish`
3- Logout and close the dialog
4- In the IDE dialog click on `Try Again`

- Repeat steps 1 -> 4 a bunch of times. Each time memory grows by 100m or so.

**Comment by ronblum** [comment]
Running the same version of RStudio (1.1.412) but on MacOS 10.12.6, I was only able to reproduce similar behavior, but not exactly the same. The base memory footprint was around 300 MB. As I went through the steps described above, the memory used kept rising (and sometimes falling a bit), till it maxed out around 600 MB. Eventually, though, after I stopped, the memory usage dropped back to around 300 MB. 

Could there be a difference between MacOS 10.11.5 and 10.12.6?
